src/index.ts

```diff
@@ -34,55 +26,51 @@ class SeqTransport extends TransportStream {
  * @param {Function} next - Continuation to respond to when complete.
  * @returns {void}
  */
-  log (info: unknown, next: () => void): void {
-    const { level, message, ...rest } =
-    <{ level:string, message: string }>info
+  log (info: any, next: () => void): void {
+    setImmediate(() => this.emit('logged', info))
+
+    // trim info to avoid passing two copies of built-in properties
+    info.exception = info.exception ? info.stack : undefined
+    const message = info.message || ''
+    delete info.message // this will be in the @MessageTemplate anyway
 
-    const timestamp = new Date()
-    setImmediate(() => {
-      this.logger.emit({
-        timestamp,
-        level: this.mapLevel(level),
-        messageTemplate: message,
-        properties: rest,
-        /*
-          exception:
-            Unsure if this can be used 🤷‍♂️
-            super has a handleExceptions option 💡
-        */
-      })
+    // differentiate events that come from handleExceptions: true or handleRejections: true
+    if (!!message && typeof message === 'string') {
+      info.winstonLogTrigger = message.startsWith('uncaughtException')
+        ? 'uncaughtException'
+        : message.startsWith('unhandledRejection')
+          ? 'unhandledRejection'
+          : undefined
+    }
 
-      setImmediate(() => this.emit('logged', info))
-    })
+    this.logger.emit({
+      timestamp: info.timestamp || new Date(),
+      level: info.level,
+      messageTemplate: message,
+      properties: info,
+    });
```

Comment:
should `exception:` be handled specifically here?

Reply:
Just thinking some more, could we do something like this to avoid duplicating the "special" properties (or some of them) in the payload?

```js
const { timestamp, level, message, exception, stack, ...props } = info;
```

and then use `properties: props` for the arg to emit?